exasol_transformers_extension/udfs/models/model_downloader_udf.py

```diff
@@ -22,11 +24,11 @@ def __init__(self,
     def run(self, ctx) -> None:
         while True:
```

Comment:
a for loop may would be more obvious and less error prone

Reply:
if you can tell me how? The thing is, I would need an until loop which doesn't exist in python.

Reply:
Something like this:

```python
class ContextIterator:

    def __init__(self, context):
        self._ctx = context

    def __iter__():
        self._ctx.next(reset=True) 

    def __next__():
        self._ctx.next(first=True)
        return self._ctx

    def __getitem__(self, key):
        return self._ctx[key]

    def __getattr__(self, key):
        return self._ctx[key]


def run(self, ctx) -> None:
    for c in ContextIterator(ctx):
        paths = self._download_model(ctx)
        c.emit(*paths)
```

An iterator like this also could be made available in general.